src/App.tsx

```diff
@@ -3,33 +3,46 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string,
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
 
   render() {
+    const { query } = this.state;
+    const visibleMovies = moviesFromServer.filter(
+      movie => movie.title.toLowerCase().includes(query.toLowerCase())
+    || movie.description.toLowerCase().includes(query.toLowerCase()),
+    );
```

Comment:
do not write logic in render, create a method;
also create for `query.toLowerCase` a variable on that method, no need to calculate it on every iteration